dev/TreeView/TreeView.cpp

```diff
@@ -38,24 +38,44 @@ TreeViewList* TreeView::ListControl()
 
 winrt::IInspectable TreeView::ItemFromContainer(winrt::DependencyObject const& container)
 {
-    return ListControl()->ItemFromContainer(container);
+    return ListControl() ? ListControl()->ItemFromContainer(container) : nullptr;
 }
 
 winrt::DependencyObject TreeView::ContainerFromItem(winrt::IInspectable const& item)
 {
-    return ListControl()->ContainerFromItem(item);
+    return ListControl() ? ListControl()->ContainerFromItem(item) : nullptr;
 }
 
 winrt::TreeViewNode TreeView::NodeFromContainer(winrt::DependencyObject const& container)
 {
-    return ListControl()->NodeFromContainer(container);
+    return ListControl() ? ListControl()->NodeFromContainer(container) : nullptr;
 }
 
 winrt::DependencyObject TreeView::ContainerFromNode(winrt::TreeViewNode const& node)
 {
-    return ListControl()->ContainerFromNode(node);
+    return ListControl() ? ListControl()->ContainerFromNode(node) : nullptr;
 }
 
+void TreeView::SelectedNode(winrt::TreeViewNode const& node)
+{
+    auto selectedNodes = SelectedNodes();
+    if (selectedNodes.Size() > 0)
+    {
+        selectedNodes.Clear();
+    }
+    if (node)
+    {
+        selectedNodes.Append(node);
+    }
+}
+
+winrt::TreeViewNode TreeView::SelectedNode()
+{
+    auto nodes = SelectedNodes();
+    return nodes.Size() > 0 ? nodes.GetAt(0) : nullptr;
+}
+
```

Comment:
nit: perhaps return an empty list here instead of nullptr?

Reply:
This is the singular version for single selection. SelectedNodes and SelectedItems do return empty lists if there's nothing selected.